### PR TITLE
Design-Pattren-project01

### PR DESCRIPTION
**Pull request by AbdulrahmanS1**
with this homeWork, I got a clear picture of the builder

### CODE REVIEW

File: Builder.cs [hunk 1]
```diff
@@ -0,0 +1,173 @@
+using System;
+using System.Collections.Generic;
+
+namespace Builder
+{
+    public interface IPizzaBuilder
+    {   
+        public void Reset();
+        public void BuildDough(string? userInput);
```
Comment (SN-SAFCSP) [inline Builder.cs:9]: Great Job using input from user!

File: Builder.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        }
+        public void Reset()
+        {
+            this.Pizza = new WheatPizza();
+        }
+        public void BuildDough(string userInput = null)
+        {
+            if (userInput == null || userInput.Trim().Length < 1)
+            {
+                this.Pizza.Add("Wheat Dough");
```
Comment (SN-SAFCSP) [inline Builder.cs:29]: It Is a great Job that you used a default value that can help the client if they didn't know what to choose 👏👏

File: Program.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+namespace Builder
+{
+   class Program
+    {
+        static void Main(string[] args)
+        {
+            var director = new Director();
+            var builder = new WhitePizzaBuilder();
+            director.Builder = builder;
+            //Bonus 2 Director takes input from user
```
Comment (SN-SAFCSP) [inline Program.cs:12]: Great Job merging the second bonus with the main requirement

File: Program.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+            director.makePizza();
+            System.Console.WriteLine("With the director");
+            Console.WriteLine(builder.GetPizza().listComponents());
+            System.Console.WriteLine("\n******************\n");
+            //Bonus 1 Custom Pizza without director
```
Comment (SN-SAFCSP) [inline Program.cs:17]: Great Job doing the first Bonus :clap:

**Review by SN-SAFCSP — COMMENTED** [review]
**Amazing Job Abdulrahman!**
You have done well, completing the assigned project and doing the extra Bonuses!
Your code is clean and organized
I think your code is Identical to another trainee. have you worked together to finish this project?! 
Thank you.
> don't forget to check the comments on your code

**Comment by SN-SAFCSP** [comment]
**Amazing Job Abdulrahman!**
You have done well, completing the assigned project and doing the extra Bonuses!
Your code is clean and organized
I think your code is Identical to another trainee. have you worked together to finish this project?! 
Thank you.
> don't forget to check the comments on your code

**Comment by AbdulrahmanS1** [comment]
> **Amazing Job Abdulrahman!**
> You have done well, completing the assigned project and doing the extra Bonuses!
> Your code is clean and organized
> I think your code is Identical to another trainee. have you worked together to finish this project?!
> Thank you.
> 
> > don't forget to check the comments on your code

To be honest I have copied one of the guys' codes, cuz of my laziness. But I didn't copy it without understanding, and I even tried to make it fail and did some test what if I remove this part and that part.
I can redo it again without using another code and am sorry for doing this.

**Comment by SN-SAFCSP** [comment]
> > **Amazing Job Abdulrahman!**
> > You have done well, completing the assigned project and doing the extra Bonuses!
> > Your code is clean and organized
> > I think your code is Identical to another trainee. have you worked together to finish this project?!
> > Thank you.
> > > don't forget to check the comments on your code
> 
> To be honest I have copied one of the guys' codes, cuz of my laziness. But I didn't copy it without understanding, and I even tried to make it fail and did some test what if I remove this part and that part.
> I can redo it again without using another code and am sorry for doing this.

Thank you for being honest. Its okay you don't need to redo it, as long as you understood the concept, but next time try to solve it yourself it would be fun!

**Pull request closed without merging**